tests/contrib/data/url2purl.json

```diff
@@ -234,21 +226,20 @@
   "https://bitbucket.org/TG1999/first_repo/src/master/new_folder/": "pkg:bitbucket/tg1999/first_repo@master#new_folder",
   "https://bitbucket.org/TG1999/first_repo/src/MASTER/NEW_FOLDER/": "pkg:bitbucket/tg1999/first_repo@MASTER#NEW_FOLDER",
   "https://bitbucket.org/TG1999/first_repo/new_folder/": "pkg:bitbucket/tg1999/first_repo@new_folder",
-  "https://bitbucket.org/multicoreware/x265/downloads/x265_2.6.tar.gz" : "pkg:bitbucket/multicoreware/x265?download_url=https://bitbucket.org/multicoreware/x265/downloads/x265_2.6.tar.gz",
-  "https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip" : "pkg:bitbucket/robeden/trove?download_url=https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip",
-
+  "https://bitbucket.org/multicoreware/x265/downloads/x265_2.6.tar.gz": "pkg:bitbucket/multicoreware/x265?download_url=https://bitbucket.org/multicoreware/x265/downloads/x265_2.6.tar.gz",
+  "https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip": "pkg:bitbucket/robeden/trove?download_url=https://bitbucket.org/robeden/trove/downloads/trove-3.0.3.zip",
   "https://gitlab.com/TG1999/firebase/-/tree/1a122122/views": "pkg:gitlab/tg1999/firebase@1a122122#views",
   "https://gitlab.com/tg1999/firebase": "pkg:gitlab/tg1999/firebase",
   "https://gitlab.com/TG1999/firebase/-/": "pkg:gitlab/tg1999/firebase",
   "https://gitlab.com/TG1999/firebase/-/tree": "pkg:gitlab/tg1999/firebase",
-  "https://gitlab.com/TG1999/firebase/-/master": "pkg:gitlab/tg1999/firebase@master",
+  "https://gitlab.com/TG1999/firebase/-/master": "pkg:gitlab/tg1999/firebase",
```

Comment:
May this be a regression?

Reply:
@pombredanne I think this is fine. I don't think `https://gitlab.com/TG1999/firebase/-/master` is a proper URL to a gitlab repo branch. 

I've tested this out by creating my own gitlab repo (https://gitlab.com/JonoYang/test) with a branch named `main`, then going to the URL `https://gitlab.com/JonoYang/test/-/main`. I get a 404 when I go to `https://gitlab.com/JonoYang/test/-/main`.